### PR TITLE
Update events-that-trigger-workflows.md

### LINKED ISSUE #23909: [REST] Activity Types list for pull_request appears stale
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

### What part(s) of the article would you like to see updated?

The list of activity types appears stale:

- assigned
- unassigned
- labeled
- unlabeled
- opened
- edited
- closed
- reopened
- synchronize
- converted_to_draft
- ready_for_review
- locked
- unlocked
- review_requested
- review_request_removed
- auto_merge_enabled
- auto_merge_disabled

At least `milestoned` is missing...

### Additional information

https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request

assigned
auto_merge_disabled
auto_merge_enabled
closed
converted_to_draft
demilestoned
dequeued
edited
enqueued
labeled
locked
milestoned
opened
ready_for_review
reopened
review_request_removed
review_requested
synchronize
unassigned
unlabeled
unlocked

image

### PR DESCRIPTION
**Pull request by mark-mxwl**
### Why:

Closes: #23909 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Updated activity types for `pull_request` to reflect webhook payload object defined [here](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request), as requested in the linked issue.

Added `enqueued` and `dequeued` to the list. Changed position of `ready_for_review`. Removed unnecessary spaces.

### Check off the following:

- [X] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [X] For content changes, I have completed the [self-review checklist](https://docs.github.com/en/contributing/collaborating-on-github-docs/self-review-checklist).

### CODE REVIEW

**Comment by mark-mxwl** [comment]
Thx @nguyenalex836!

File: content/actions/using-workflows/events-that-trigger-workflows.md [hunk 1]
```diff
@@ -586,7 +586,7 @@ on:
...
+| [`pull_request`](/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request) | - `assigned`<br/>- `unassigned`<br/>- `labeled`<br/>- `unlabeled`<br/>- `opened`<br/>- `edited`<br/>- `closed`<br/>- `reopened`<br/>- `synchronize`<br/>- `converted_to_draft`<br/>- `locked`<br/>- `unlocked`<br/>- `enqueued`<br/>- `dequeued`<br/>- `milestoned`<br/>- `demilestoned`<br/>- `ready_for_review`<br/>- `review_requested`<br/>- `review_request_removed`<br/>- `auto_merge_enabled`<br/>- `auto_merge_disabled` | Last merge commit on the `GITHUB_REF` branch | PR merge branch `refs/pull/PULL_REQUEST_NUMBER/merge` |
```
Comment (jc-clark) [inline content/actions/using-workflows/events-that-trigger-workflows.md:589]: While we were looking into this, we noticed that the `enqueued` and `dequeued` activity types aren't listed in the GitHub Enterprise Server versions of the webhooks article. 

You can see that here: 

https://docs.github.com/en/enterprise-server@3.11/webhooks/webhook-events-and-payloads#pull_request

What do you think about adding some versioning here to only show these types for the Free, Pro, Team and the GitHub Enterprise Cloud versions of the article? 

```suggestion
| [`pull_request`](/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request) | - `assigned`<br/>- `unassigned`<br/>- `labeled`<br/>- `unlabeled`<br/>- `opened`<br/>- `edited`<br/>- `closed`<br/>- `reopened`<br/>- `synchronize`<br/>- `converted_to_draft`<br/>- `locked`<br/>- `unlocked`<br/>{% ifversion fpt or ghec %}- `enqueued`<br/>- `dequeued`<br/>{% endif %}- `milestoned`<br/>- `demilestoned`<br/>- `ready_for_review`<br/>- `review_requested`<br/>- `review_request_removed`<br/>- `auto_merge_enabled`<br/>- `auto_merge_disabled` | Last merge commit on the `GITHUB_REF` branch | PR merge branch `refs/pull/PULL_REQUEST_NUMBER/merge` |
```

**Review by jc-clark — COMMENTED** [review]
Thank you for this contribution and the help to keep this article updated @mark-mxwl! 

I added a comment about versioning (which is always so tricky 😅) Let me know what you think!

File: content/actions/using-workflows/events-that-trigger-workflows.md (hunk 1 above)
Comment (mark-mxwl) [reply content/actions/using-workflows/events-that-trigger-workflows.md:589]: Great catch! 😅 And yes, I agree that adding a conditional there is a nice fix. Thanks @jc-clark!

**Comment by jc-clark** [comment]
Added this to the merge queue! Thanks again @mark-mxwl! ✨

**Comment by mark-mxwl** [comment]
Np! 🙌🏻

**Review by jc-clark — APPROVED** [review]

**Pull request merged**